k2/csrc/host/fsa_equivalent.cc

```diff
@@ -288,8 +285,6 @@ bool IsRandEquivalent(const Fsa &a, const Fsa &b,
     if (cost_a < loglike_cutoff_a && cost_b < loglike_cutoff_b) continue;
```

Comment:
As we allow epsilon arc in path, I believe `RandomPath(fsa, false, &valid_path_storage)` will never fail. So is that better to add a value `tries` here for `if (cost_a < loglike_cutoff_a && cost_b < loglike_cutoff_b) continue` and return true with a warning LOG when, e.g. `tries > 50`. I just feel the default `npath = 100` now will not be a good default value for most of large size Fsas.

Reply:
Mm the right fix here, I think, is to use a rand-path algorithm that does not generate paths outside the beam.  I dont want changes that will make this super slow, thats why I did it like this.